src/pytorch_lightning/trainer/trainer.py

```diff
@@ -637,6 +641,13 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
                 return self.strategy.launcher.launch(trainer_fn, *args, trainer=self, **kwargs)
             else:
                 return trainer_fn(*args, **kwargs)
+
+        except _TunerExitException as exception:
+            self.state.status = TrainerStatus.FINISHED
+            self._call_callback_hooks("on_exception", exception)
```

Comment:
I interpret `on_exception` as a hook to inject logic when an exception is raised that messes up your run.

But this `_TunerExitException` is not really an exception in the sense that something went wrong. You are just using it to bypass everything and stop.

Could we avoid using an exception and instead do `loop.should_stop`? Then training would stop normally

Reply:
answered here: https://github.com/Lightning-AI/lightning/pull/11089/files#r972909982